metadatastore/commands.py

```diff
@@ -211,76 +616,53 @@ def insert_run_stop(run_start, time, uid=None, exit_status='success',
 
     Returns
     -------
-    run_stop : mongoengine.Document
-        Inserted mongoengine object
+    run_stop : str
+        uid of inserted Document
     """
-    if uid is None:
-        uid = str(uuid.uuid4())
     if custom is None:
         custom = {}
-    run_start = _get_mongo_document(run_start, RunStart)
-    run_stop = RunStop(run_start=run_start, reason=reason, time=time,
-                       time_as_datetime=_todatetime(time), uid=uid,
+    runstart_uid = doc_or_uid_to_uid(run_start)
+    run_start = runstart_given_uid(runstart_uid)
+    try:
+        runstop_by_runstart(runstart_uid)
+    except NoRunStop:
+        pass
+    else:
+        raise RuntimeError("Runstop already exits for {!r}".format(run_start))
+    # look up the oid
+    runstart_oid = _RUNSTART_UID_to_OID_MAP[runstart_uid]
+    # create a reference field
+    rs_ref = DBRef('RunStart', runstart_oid)
+
+    run_stop = RunStop(run_start=rs_ref, reason=reason, time=time,
+                       uid=uid,
                        exit_status=exit_status, **custom)
 
-    run_stop.save(validate=True, write_concern={"w": 1})
+    run_stop = run_stop.save(validate=True, write_concern={"w": 1})
+    _cache_runstop(run_stop.to_mongo().to_dict())
     logger.debug("Inserted RunStop with uid %s referencing RunStart "
-                 " with uid %s", run_stop.uid, run_start.uid)
+                 " with uid %s", run_stop.uid, run_start['uid'])
 
     return uid
 
 
 @_ensure_connection
-def insert_beamline_config(config_params, time, uid=None):
-    """ Create a beamline_config  in metadatastore database backend
-
-    Parameters
-    ----------
-    config_params : dict
-        Name/value pairs that indicate beamline configuration
-        parameters during capturing of data
-    time : float
-        The date/time as found at the client side when the
-        beamline configuration is created.
-    uid : str, optional
-        Globally unique id string provided to metadatastore
-
-    Returns
-    -------
-    blc : BeamlineConfig
-        The document added to the collection
-    """
-    if uid is None:
-        uid = str(uuid.uuid4())
-    beamline_config = BeamlineConfig(config_params=config_params,
-                                     time=time,
-                                     uid=uid)
-    beamline_config.save(validate=True, write_concern={"w": 1})
-    logger.debug("Inserted BeamlineConfig with uid %s",
-                 beamline_config.uid)
-
-    return uid
-
-
-@_ensure_connection
-def insert_event_descriptor(run_start, data_keys, time, uid=None,
+def insert_descriptor(run_start, data_keys, time, uid,
                             custom=None):
```

Comment:
the white space on this line is offset from the `(`. It might be possible to stick `custom=None` on the previous line too?